### PR TITLE
New Geometry schema

### PR DESCRIPTION
**Pull request by jmcanterafonseca**

### CODE REVIEW

**Comment by jmcanterafonseca** [comment]
@juliozinga can you have a look as well, as you opened a PR initially

**Comment by jmcanterafonseca** [comment]
@chicco785 please could your r? we need to land this asap

File: specs/UrbanMobility/GtfsShape/schema.json [hunk 1]
```diff
@@ -55,4 +43,4 @@
     "type",
     "location"
   ]
-}
+}
```
Comment (chicco785) [inline specs/UrbanMobility/GtfsShape/schema.json]: missing CR

File: geometry-schema.json [hunk 2]
```diff
@@ -1,91 +1,224 @@
...
-            "properties": {
-                "type": { "enum": [ "Point" ] },
-                "coordinates": { "$ref": "#/definitions/position" }
-            }
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "http://geojson.org/schema/Geometry.json",
+  "title": "GeoJSON Geometry",
+  "oneOf": [
+    {
+      "$id": "#Point",
```
Comment (chicco785) [inline geometry-schema.json]: internal $id is not used in the official schema, the rather provide a specific document per type to point directly to it.

File: geometry-schema.json [hunk 3]
```diff
@@ -1,91 +1,224 @@
...
+          "type": "array",
+          "minItems": 4,
+          "items": {
+            "type": "number"
+          }
+        }
+      }
+    },
+    {
+      "$id": "#Polygon",
```
Comment (chicco785) [inline geometry-schema.json]: internal $id is not used in the official schema, the rather provide a specific document per type to point directly to it.

File: geometry-schema.json [hunk 4]
```diff
@@ -1,91 +1,224 @@
...
+          "type": "array",
+          "minItems": 4,
+          "items": {
+            "type": "number"
+          }
+        }
+      }
+    },
+    {
+      "$id": "#LineString",
```
Comment (chicco785) [inline geometry-schema.json]: internal $id is not used in the official schema, the rather provide a specific document per type to point directly to it.

File: geometry-schema.json [hunk 5]
```diff
@@ -1,91 +1,224 @@
...
+          }
         }
-    ],
-    "definitions": {
-        "position": {
-            "description": "A single position",
+      }
+    },
+    {
+      "$id": "#MultiPoint",
```
Comment (chicco785) [inline geometry-schema.json]: internal $id is not used in the official schema, the rather provide a specific document per type to point directly to it.

File: geometry-schema.json [hunk 6]
```diff
@@ -1,91 +1,224 @@
...
+          "type": "array",
+          "minItems": 4,
+          "items": {
+            "type": "number"
+          }
+        }
+      }
+    },
+    {
+      "$id": "#MultiLineString",
```
Comment (chicco785) [inline geometry-schema.json]: internal $id is not used in the official schema, the rather provide a specific document per type to point directly to it.

File: geometry-schema.json [hunk 7]
```diff
@@ -1,91 +1,224 @@
...
+          "type": "array",
+          "minItems": 4,
+          "items": {
+            "type": "number"
+          }
+        }
+      }
+    },
+    {
+      "$id": "#MultiPolygon",
```
Comment (chicco785) [inline geometry-schema.json]: internal $id is not used in the official schema, the rather provide a specific document per type to point directly to it.

File: specs/UrbanMobility/GtfsShape/schema.json [hunk 8]
```diff
@@ -24,28 +26,14 @@
...
-              "properties": {
-                "type": {
-                  "type": "string",
-                  "enum": ["LineString"]
-                },
-                "coordinates": { "$ref": "http://json-schema.org/geojson/geometry.json#/definitions/lineString" }
-              }
+          "oneOf": [
+            {
+              "$ref": "http://geojson.org/schema/Geometry.json#LineString"
```
Comment (chicco785) [inline specs/UrbanMobility/GtfsShape/schema.json]: this cannot be referenced using actual url

File: specs/UrbanMobility/GtfsShape/schema.json [hunk 9]
```diff
@@ -24,28 +26,14 @@
...
-                    "type": "string",
-                    "enum": ["MultiLineString"]
-                  },
-                  "coordinates": {
-                    "type": "array",
-                    "items": { "$ref": "http://json-schema.org/geojson/geometry.json#/definitions/lineString" }
-                  }
-                }
-           }]
+              "$ref": "http://geojson.org/schema/Geometry.json#MultiLineString"
```
Comment (chicco785) [inline specs/UrbanMobility/GtfsShape/schema.json]: this cannot be referenced using actual url

File: geometry-schema.json (hunk 2 above)
Comment (jmcanterafonseca) [reply geometry-schema.json]: yes, that's why I've opened https://github.com/geojson/schema/issues/10

File: specs/UrbanMobility/GtfsShape/schema.json (hunk 9 above)
Comment (jmcanterafonseca) [reply specs/UrbanMobility/GtfsShape/schema.json]: It depends on how you map that URL to an actual schema. I don't like to have individual schemas per type of GeoJSON Geometry I think it does not make sense

Comment (chicco785) [reply specs/UrbanMobility/GtfsShape/schema.json]: URL is by definition something that you resolve :) So while I understand your point, I disagree to the "hack". The solution is getting approved the change in GeoJSON and meanwhile adhere to what they offer.

File: geometry-schema.json (hunk 2 above)
Comment (chicco785) [reply geometry-schema.json]: i like having $id, what I don't like is not hacking the GeoJSON schema ad hoc.

Comment (jmcanterafonseca) [reply geometry-schema.json]: but that' something needed, if the GeoJSON Repo people do not respond we cannot be blocked forever ...

File: specs/UrbanMobility/GtfsShape/schema.json (hunk 9 above)
Comment (jmcanterafonseca) [reply specs/UrbanMobility/GtfsShape/schema.json]: well, they are URIs not URLs ... https://json-schema.org/latest/json-schema-core.html#rfc.section.8.2 so not necessarily dereferenceable

Comment (chicco785) [reply specs/UrbanMobility/GtfsShape/schema.json]: look, I am not stubborn on this, I just believe that not creating misalignments is the good way to go :)

**Comment by jmcanterafonseca** [comment]
@chicco785 can we just land this with the commitment that if finally our change is not accepted by the GeoJSON Community, it will be aligned with whatever they produce?

thanks!

**Pull request merged**